src/psij/executors/rest_service.py

```diff
@@ -0,0 +1,163 @@
+"""This module contains the RestService :class:`~psij.JobExecutor`."""
+
+import json
+import requests
+import websocket
+import threading
+
+from urllib.parse import urlparse, urlunparse
+
+from typing import Optional, List, Dict
+
+from psij import (
+    Job,
+    JobExecutorConfig,
+    JobState,
+    JobStatus,
+    JobExecutor,
+    Export,
+)
+
+
+class RestServiceJobExecutor(JobExecutor):
+    """A :class:`~psij.JobExecutor` for a PSI/J Rest service endpoint.
+
+    This executor forwards all requests to a Rest service endpoint which is then
+    executing the respective request on the target resource.
+    """
+
+    _state_map = {'NEW': JobState.NEW,
+                  'QUEUED': JobState.QUEUED,
+                  'ACTIVE': JobState.ACTIVE,
+                  'COMPLETED': JobState.COMPLETED,
+                  'FAILED': JobState.FAILED,
+                  'CANCELED': JobState.CANCELED}
+    _final = [JobState.COMPLETED, JobState.FAILED, JobState.CANCELED]
+
+    def __init__(
+        self, url: Optional[str] = None, config: Optional[JobExecutorConfig] = None
+    ) -> None:
+        """
+        Initializes a `RestServiceJobExecutor`.
+
+        :param url: address at which to contact the remote service.
+                    Supported schemas: `http://` and `https://`
+        :param config: The `RestServiceJobExecutor` does not have any
+                    configuration options.
+        """
+        if not config:
+            config = JobExecutorConfig()
+
+        if not url:
```

Comment:
Maybe better to require a url?

Reply:
Yeah, that's fair - I'll remove the default.

Reply:
Looks good although I think instead of checking if not url you could just make it a required/positional argument.